lib/kennel.rb

```diff
@@ -126,12 +89,18 @@ def api
       @api ||= Api.new(ENV.fetch("DATADOG_APP_KEY"), ENV.fetch("DATADOG_API_KEY"))
     end
 
+    def projects_generator
+      @projects_generator ||= ProjectsGenerator.new
```

Comment:
how about "loader" ?
```suggestion
      @projects_generator ||= ProjectsLoader.new
```

Reply:
Well if anything it's more like a `Provider`. It's a thing which can provide a set of projects. This one just happens to do it by using Zeitwerk / load_all and then doing the all-subclasses-of-Project thing. The intention is that, for testing, it will be (i.e. in future PRs) useful to supply other ways of providing projects (e.g. "here is an array of projects"), because then that's much easier to work with for testing. You don't need temp dirs, you don't need to write to the filesystem, you don't need to unload classes and mess around with `Object.send(:remove_const)`.

But having said that: it's just a name, so we can call it `Loader` if you like and still do the testing stuff I just said.

Reply:
dd9883e62fcb84fe3f79879525d52c47161f5c61